tests/test_tracing.py

```diff
@@ -1,13 +1,68 @@
 import os
 import time
 
+import opentelemetry.exporter.otlp.proto.http.trace_exporter
 from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.trace.export import ConsoleSpanExporter
 
 from jobrunner import config, models, tracing
 from tests.conftest import get_trace
 from tests.factories import job_factory, job_request_factory, job_results_factory
 
 
+def test_setup_default_tracing_empty_env(monkeypatch):
+    env = {}
+    monkeypatch.setattr(os, "environ", env)
+    provider = tracing.setup_default_tracing(set_global=False)
+    assert provider._active_span_processor._span_processors == ()
+
+
+def test_setup_default_tracing_console(monkeypatch):
+    env = {"OTEL_EXPORTER_CONSOLE": "1"}
+    monkeypatch.setattr(os, "environ", env)
+    provider = tracing.setup_default_tracing(set_global=False)
+
+    processor = provider._active_span_processor._span_processors[0]
+    assert isinstance(processor.span_exporter, ConsoleSpanExporter)
+
+
+def test_setup_default_tracing_otlp_defaults(monkeypatch):
+    env = {"OTEL_EXPORTER_OTLP_HEADERS": "foo=bar"}
+    monkeypatch.setattr(os, "environ", env)
+    monkeypatch.setattr(
+        opentelemetry.exporter.otlp.proto.http.trace_exporter, "environ", env
+    )
```

Comment:
just curious - why doesn't monkeypatching `os` also change it here? Is it just because `opentelemetry.exporter.otlp.proto.http.trace_exporter` does `from os import environ` rather than just `import os`, or is there something else going on?

Reply:
That is exactly what is going on, sadly.